tests/Unit/Elliptic/Actions/Test_InitializeSystem.cpp

```diff
@@ -0,0 +1,258 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Creators/Brick.hpp"
+#include "Domain/Creators/Interval.hpp"
+#include "Domain/Creators/Rectangle.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/Actions/InitializeSystem.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp"
+#include "NumericalAlgorithms/LinearSolver/Tags.hpp"
+#include "Parallel/AddOptionsToDataBox.hpp"
+#include "ParallelAlgorithms/DiscontinuousGalerkin/InitializeDomain.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace {
+
+struct ScalarFieldTag : db::SimpleTag {
+  static std::string name() noexcept { return "ScalarFieldTag"; };
+  using type = Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct System {
+  static constexpr size_t volume_dim = Dim;
+  using fields_tag = Tags::Variables<tmpl::list<ScalarFieldTag>>;
+  using gradient_tags = tmpl::list<LinearSolver::Tags::Operand<ScalarFieldTag>>;
+};
+
+template <size_t Dim>
+struct AnalyticSolution {
+  tuples::TaggedTuple<Tags::Source<ScalarFieldTag>> variables(
+      const tnsr::I<DataVector, Dim>& x,
+      tmpl::list<Tags::Source<ScalarFieldTag>> /*meta*/) const noexcept {
+    return {Scalar<DataVector>(get<0>(x))};
```

Comment:
Can you add a dependence on `y` and `z`?

Reply:
done, please check the fixup

Reply:
Looks good! Please go ahead and squash :)